src/main/java/org/mapstruct/intellij/util/InheritConfigurationUtils.java

```diff
@@ -0,0 +1,204 @@
+    }
+
+    private static List<PsiMethod> findLocalAndParentMethods(PsiClass containingClass) {
+
+        List<PsiMethod> result = new ArrayList<>( List.of( containingClass.getMethods() ) );
+
+        for ( PsiClass anInterface : containingClass.getInterfaces() ) {
+            result.addAll( findLocalAndParentMethods( anInterface ) );
+        }
+
+        return result;
+    }
+
+    private static Stream<PsiMethod> findMapperConfigMethods(@NotNull PsiAnnotation mapperAnnotation) {
+
+        PsiModifierListOwner mapperConfigReference = findMapperConfigReference( mapperAnnotation );
+
+        if ( !( mapperConfigReference instanceof PsiClass ) ) {
+            return Stream.empty();
+        }
+
+        return Arrays.stream( ( (PsiClass) mapperConfigReference ).getMethods() );
+    }
+
+    /**
+     * Find a candidate that this mapping method can inherit from, only if it is the only one possible.
+     *
+     * @param mappingMethod the mapping method to find a belonging method
+     * @param candidates mapping methods that possibly hold the method to inherit from
+     * @param inheritConfigurationAnnotation the {@link org.mapstruct.InheritConfiguration} annotation
+     * @return a mapping method to inherit from, only if it is the only possible one found.
+     */
+    public static Optional<PsiMethod> findSingleMatchingInheritMappingMethod(
+        @NotNull PsiMethod mappingMethod,
+        @NotNull Stream<PsiMethod> candidates,
+        @NotNull PsiAnnotation inheritConfigurationAnnotation) {
+
+        String inheritConfigurationName = getStringAttributeValue( inheritConfigurationAnnotation, "name" );
+
+        List<PsiMethod> matchingCandidates = candidates
+            .filter( candidate -> isNotTheSameMethod( mappingMethod, candidate ) )
+            .filter( candidate -> matchesNameWhenNameIsDefined( inheritConfigurationName, candidate ) )
+            .filter( candidate -> canInheritFrom( mappingMethod, candidate ) )
+            .collect( Collectors.toList() );
+
+        if ( matchingCandidates.size() != 1 ) {
+            return Optional.empty();
+        }
+
+        return matchingCandidates.stream().findFirst();
+    }
+
+    private static boolean isNotTheSameMethod(PsiMethod mappingMethod, PsiMethod candidate) {
+
+        return !( mappingMethod.equals( candidate ) );
+    }
+
+    private static boolean matchesNameWhenNameIsDefined(String inheritConfigurationName, PsiMethod candidate) {
+
+        if ( inheritConfigurationName == null || inheritConfigurationName.isEmpty() ) {
+            return true;
+        }
+
+        return candidate.getName().equals( inheritConfigurationName );
+    }
+
+    /**
+     * simplified version of <code>org.mapstruct.ap.internal.model.source.SourceMethod#canInheritFrom</code>
+     */
+    public static boolean canInheritFrom(PsiMethod mappingMethod, PsiMethod candidate) {
+
+        PsiType targetType = findTargetTypeOfMappingMethod( mappingMethod );
```

Comment:
Slight optimization. Instead of getting the target type every time, why not pass it to the `canInheritMethod` instead?